src/commands/org/display.ts

```diff
@@ -101,11 +102,15 @@ export class OrgDisplayCommand extends SfCommand<OrgDisplayReturn> {
     });
   }
 
-  private async getScratchOrgInformation(orgId: string): Promise<ScratchOrgFields> {
+  private async getScratchOrgInformation(orgId: string, username: string): Promise<ScratchOrgFields> {
     const hubOrg = await this.org.getDevHubOrg();
     // we know this is a scratch org so it must have a hubOrg and that'll have a username
     const hubUsername = hubOrg?.getUsername() as string;
-    const result = (await OrgListUtil.retrieveScratchOrgInfoFromDevHub(hubUsername, [trimTo15(orgId)]))[0];
+    // this query can return multiple records that match the 15 char ID because `ScratchOrgInfo.ScratchOrg` isn't a case-sensitive field
+    // so we look for the record that matches the scratch org username in the auth file.
+    const result = (await OrgListUtil.retrieveScratchOrgInfoFromDevHub(hubUsername, [trimTo15(orgId)])).find(
+      (rec) => rec.SignupUsername === username
```

Comment:
`OrgListUtil.retrieveScratchOrgInfoFromDevHub` is also used by `org list` but it's not a problem there because table rows data is generated from each payload.

`org display` builds the info object for one org by merging auth fields and the payload response so we need to be sure it's the correct record here.